src/DataService/Annotation/DataService.php

```diff
@@ -59,10 +50,16 @@ class DataService
     /**
      * Return the service of the data object, such as 'Media Data Service'.
      *
-     * @return string
+     * @param bool $readonly (optional) Set to true to return the read-only name of the service.
+     *
+     * @return string The name of the service.
      */
-    public function getService(): string
+    public function getService($readonly = false): string
```

Comment:
This is a boolean so if later we find that appending a string doesn't work that we can have whatever logic we need (such as pulling it from an annotation property).